libmscore/segment.cpp

```diff
@@ -1029,9 +1029,11 @@ void Segment::scanElements(void* data, void (*func)(void*, Element*), bool all)
                   e->scanElements(data, func, all);
                   }
       for (Element* e : annotations()) {
-            if (all || e->systemFlag() || measure()->visible(e->staffIdx()))
-                  e->scanElements(data,  func, all);
-            }
+             if (score()->tagIsValid(e->tag())) {
```

Comment:
Yes, so that only the elements with the appropriate tags (depending upon the layer selected) are drawn.